examples/example-next-13-next-auth/src/app/layout.tsx

```diff
@@ -1,12 +1,17 @@
 import {ReactNode} from 'react';
+import {useLocale, useLanguageDirection} from 'next-intl';
 
 type Props = {
   children: ReactNode;
 };
 
-// Even though this component is just passing its children through, the presence
-// of this file fixes an issue in Next.js 13.3.0 where link clicks that switch
-// the locale would otherwise be ignored.
-export default function RootLayout({children}: Props) {
-  return children;
+export default function Layout({children}: Props) {
+  const locale = useLocale();
```

Comment:
As a side note, `useLocale` will likely not work outside of the `[locale]` folder when static rendering support lands in next-intl based on `createServerContext`.

Reply:
If `useLocale` will not work outside the context, `useTextDirection` depends on the same approach so it won't work either.